### ISSUE TITLE
Camera Drone Controls

### ISSUE THREAD

**Issue by Brusura** [Okabintaro/SubmersedVR#24]
I am using Quest 2 with Virtual Desktop e SteamVR , I can not switch drones, the icon tips tell me to press left and right but I do not have those keys on Quest 2 controllers

Also the drones move always in the same direction independet of whare you are looking at

**Comment by hearth1an** [comment]
I have the same issue

**Comment by benjamincheng10** [comment]
yes i have the same issue, please make the cameras follow the player direction, and remap input to allow switching cameras please!

**Comment by RichardDDB** [comment]
I have the same issues, including the drones changing the direction their facing after taking a hit.

**Comment by realgravy** [comment]
I just saw this and this is not an issues. The Cycle Previous\Next button is not mapped to any controller buttons. Go into Steam Configuration and set a button - I used analog click personally.

**Comment by Jyers** [comment]
Just wanted to chime in that issue of drones not following head position is still very much present. Now its not even just caused by collisions but happens at all times. I'm usually pretty good with motion sickness, but attempting to use drones is nauseating currently. @Okabintaro if you don't have time to fix this, I'd happily take a stab at it. I might just need pointing in the right direction as to where in the code this would be handled.

**Comment by Jyers** [comment]
There is also an alternative. If the UI for the drone were to stay stationary while the player moved their head, it would give a clear frame of reference for what direction is forward and should significantly reduce motion sickness. This would make the cameras more like a vehicle which makes sense. Also, it would fix #63 since then one could simply look over to be able to see the numbers properly

**Comment by Okabintaro** [comment]
Heya, you are right that I didn't pay much attention to the drone cameras in the past, since I rarely use them myself.
I think the same confusion is present right now with the cyclops. I think ideally those cameras should be displayed as floating screens or something in VR, but making them behave more like vehicles should be fine too.
I haven't looked into the code, but for vehicles the whole player controller gets parented to the vehicle while that probably doesn't happen to the drone and cyclops cameras.

Hopefully I will have some time figuring it out, I plan to tackle the most serve bugs in the next few weeks.
If you want you could also try to study how it works with dnspy and/or UnityExplorer, I wouldn't really know right now how to give pointers.

**Comment by Jyers** [comment]
Oooo, yes a floating screen would also be a really cool implementation. Especially since it would make it flat rather than 3D which is truer to life for a camera. If I find some time to look into it a bit I will update here.

**Comment by KihonBlue** [comment]
Yip, I have same problem, can't switch the drone cameras as it's left/right but that doesn't work in VR with motion controllers.
Also issues with for example using the seaglide and harvesting resources by pressing the A button also switches the saeglide map on/off, ie double binding.

**Comment by KihonBlue** [comment]
> I just saw this and this is not an issues. The Cycle Previous\Next button is not mapped to any controller buttons. Go into Steam Configuration and set a button - I used analog click personally.

How do I set a button to change the camera in Steam configuration, what is the actual option, find that steam button configuration menu extremely confusing.
